NuRadioReco/modules/neutrinoEnergyReconstructor.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+import NuRadioMC.SignalProp.analyticraytracing
+from NuRadioReco.framework.parameters import channelParameters as chp
+from NuRadioReco.framework.parameters import electricFieldParameters as efp
+from NuRadioReco.framework.parameters import stationParameters as stnp
+from NuRadioReco.framework.parameters import showerParameters as shp
+import NuRadioReco.framework.base_trace
+import NuRadioReco.framework.radio_shower
+from NuRadioReco.utilities import trace_utilities, units
+
+
+class neutrinoEnergyReconstructor():
+    """
+    Module to perform an energy reconstruction as described in https://arxiv.org/abs/2107.02604
+    Look at NuRadioReco/examples/RNO_energy_reconstruction for an example how to use it.
+    Requires vertex reconstruction and IFT electric field reconstruction to be run beforehand.
+    """
+    def __init__(self):
+        self.__s_parametrization = np.array([4.70, -22.20, 29.99])
+        self.__s_prime_parametrization = np.array([6.46, -16.00, 12.49])
+        self.__s_passbands = np.array([
+            [.13, .3],
+            [.3, .5]
+        ])
+        self.__s_prime_passbands = np.array([
+            [.13, .2],
+            [.2, .3]
+        ])
+        self.__s_s_prime_cut_value = 10.
+        self.__channel_groups = None
+        self.__raytracer = None
+
+    def begin(self, channel_groups, ice, attenuation_model='GL1'):
+        """
+        Set up module
+
+        Parameters
+        -------------------
+        channel_groups: 2D array of integers
+            List of groups of channels which were used together by the iftElectricFieldReconstructor
+        ice: ice model object
+            Class describing the index of refraction model of the ice
+        attenuation model: string
+            Name of the model describing the signal attenuation
+        """
+        self.__channel_groups = channel_groups
+        self.__raytracer = NuRadioMC.SignalProp.analyticraytracing.ray_tracing(
+            ice,
+            attenuation_model
+        )
+        pass
+
+    def run(self, event, station, detector):
```

Comment:
Maybe add a docstring, not so much because of the parameters, but to indicate what is reconstructed and where it is stored (i.e. in a `shower` as `shp.energy`)